### PR TITLE
feat!: simplifying spent proofs and TX verification public APIs

### PR DESCRIPTION
**Pull request by bochaco**
This is to simplify the public API exposed by sn-dbc for Tx and `SpentProofs` verification, reducing the public `Trait` to a single function, while moving the rest (`KeyManager`) to the mock as it's not required on the public API but for the example code/app.

### CODE REVIEW

File: examples/mint-repl/mint-repl.rs [hunk 1]
```diff
@@ -85,7 +85,7 @@ fn main() -> Result<()> {
     loop {
         match rl.readline(">> ") {
             Ok(line) => {
-                let mut args = line.trim().split_whitespace();
+                let mut args = line.split_whitespace();
```
Comment (davidrusu) [inline examples/mint-repl/mint-repl.rs:88]: Does this still work when we have leading whitespace in the line input? ie. `<space>reissue`

**Review by davidrusu — APPROVED** [review]
q on the mint-repl change, but looks good otherwise

File: examples/mint-repl/mint-repl.rs (hunk 1 above)
Comment (bochaco) [reply examples/mint-repl/mint-repl.rs:88]: It's supposed to, it's the linter who suggested it: https://rust-lang.github.io/rust-clippy/master/index.html#trim_split_whitespace

Comment (davidrusu) [reply examples/mint-repl/mint-repl.rs:88]: :+1: nice

**Pull request merged**